service/internal/graph/graph.go

```diff
@@ -196,6 +196,18 @@ func (g *Graph) createNodes(set Settings) error {
 	return nil
 }
 
+// formatPipelineNameWithSignal formats pipeline name with signal as "signal[/name]" format.
+func formatPipelineNameWithSignal(pipelineIDs []pipeline.ID, signal pipeline.Signal) string {
+	var formatted string
+	for _, pid := range pipelineIDs {
+		if pid.Signal() == signal {
+			formatted = pid.String()
+			break
+		}
+	}
```

Comment:
I think you should return a slice here (since there may be more than 1), and use "%v" in the errorf.

Reply:
@sudiptob2 I think you missed this.

Reply:
O yeah! thanks for mentioning me. Updated the code now.
But we can only catch one signal type. For example with the slice method, the output will be either `[logs/in, logs/in2]` or `[traces]`. not both like `[logs/in logs/in2 traces]` for the following config file.
```yaml
  pipelines:
    logs/in:
      receivers: [otlp]
      processors: []
      exporters: [forward]
    logs/in2:
      receivers: [ otlp ]
      processors: [ ]
      exporters: [ forward ]
    logs/out:
      receivers: [otlp]
      processors: []
      exporters: [debug]
    traces:
      receivers: [ otlp ]
      processors: [ ]
      exporters: [ forward ]
    metrics:
      receivers: [ forward ]
      processors: [ ]
      exporters: [ debug ]
```

Reply:
Also, the order in which they appear in the slice is not deterministic.